script/bootstrap

```diff
@@ -34,13 +34,13 @@ if (dependenciesFingerprint.isOutdated()) {
 
 if (process.platform === 'win32') deleteMsbuildFromPath()
 
-installScriptDependencies(ci)
```

Comment:
Just set `ci=false` instead of removing it?

Reply:
I realized we need to keep the CI bits and use it for `--no-save` so `package.json` doesn't update and cause the saved cache not to match.

Once I figure it out on my fork I'll push here, unless you'd like me to do it here, I just don't like posting work that I am not confident it in the details of yet. And I can spam my personal fork with however many force pushes as I want, hehe.

Reply:
upstream has a bot that bumps the "version" field of `[repo_root]/package.json`, but not `[repo_root]/package-lock.json`. Meaning... any time you do `npm install`, the "version field of `package-lock.json` updates. (`npm ci` refuses to update `package-lock.json` in any case, that's why we didn't have that problem before.)

I think outside of CI you should know when your `package-lock.json` needs to be checked in, so using `npm install --no-save` outside of CI isn't desirable IMO. So I have this ternary operator for setting the flag only in CI:

`'npm', [`install', ci ? '--no-save' : null]`

Reply:
I meant that set it to false in JavaScript if it is needed in other places. 

One thing we can do is to set ci=true in Linux (which builds and bootstraps very fast) and set it false in other operating systems to speed up the builds.

In GitHub actions, once can use commit messages to command the CI to do various things. We might want to don't remove this altogether, and instead make a way to force a clean `npm ci` bootstrap when we need it.

Something like this: 
https://github.com/atom-ide-community/minimap-plus/blob/06253757e7700a660e3e0dd7ae0621c46a5539a5/.github/workflows/CI.yml#L85
```
    if: contains(github.event.head_commit.message, '[npm ci]')
```

I wonder if Azure has such thing

Reply:
https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#agent-variables

> `Build.SourceVersionMessage`
>
> The comment of the commit or changeset for the triggering repo.
> 
> This variable is agent-scoped, and can be used as an environment variable in a script and as a parameter in a build task, but not as part of the build number or as a version control tag. Also, this variable is only available on the step level and is neither available in the job nor stage levels (i.e. the message is not extracted until the job had started and checked out the code).